topdown/topdown.go

```diff
@@ -1898,7 +1898,8 @@ func evalTermsComprehension(t *Topdown, comp ast.Value, iter Iterator) error {
 		child := t.Closure(comp.Body)
 
 		err := Eval(child, func(child *Topdown) error {
-			result = append(result, PlugTerm(comp.Term, child.Binding))
+			t := expandTermRef(child, comp.Term)
```

Comment:
I would change this to:

1. Plug the term to replace values w/ bindings (`PlugTerm`)
2. Resolve refs to replace local var refs w/ values (`ResolveRefs`)

This should remove the need for `expandTermRef` here.